broker-daemon/utils/create-basic-auth.js

```diff
@@ -0,0 +1,44 @@
+const { PublicError } = require('grpc-methods')
+
+/**
+ * @constant
+ * @type {Boolean}
+ * @default
+ */
+const BASIC_AUTH_DELIMITER = ':'
+
+/**
+ * Verifies a username and password from metadata sent from grpc-methods. This function is
+ * called from inside the grpc-methods authorization hook.
+ *
+ * @see {@link https://github.com/sparkswap/grpc-methods}
+ * @param {String} rpcUser
+ * @param {String} rpcPass
+ * @param {Boolean} [disableAuth=false]
+ * @return {Function}
+ */
+function createBasicAuth (rpcUser, rpcPass, disableAuth = false) {
+  return async ({ metadata, logger }) => {
```

Comment:
why does this return an `async` function? It looks synchronous to me. Is this reserving optionality for future changes that may have an async credentials check (e.g. that does IO)?

Reply:
@treygriffith this is `awaited` in grpc-methods